### PR TITLE
Add compat packages to firebase-exp

### PR DESCRIPTION
**Pull request by Feiyang1**

### CODE REVIEW

File: packages-exp/auth-compat-exp/index.ts [hunk 1]
```diff
@@ -140,5 +139,4 @@ function registerAuthCompat(instance: _FirebaseNamespace): void {
   instance.registerVersion('auth', version);
 }
 
-impl.registerAuth(_getClientPlatform());
```
Comment (hsubox76) [inline packages-exp/auth-compat-exp/index.ts:143]: What happened with this, did getClientPlatform turn out to be unnecessary?

File: packages-exp/firebase-exp/compat/auth/index.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
+/**
+ * @license
+ * Copyright 2020 Google LLC
```
Comment (hsubox76) [inline packages-exp/firebase-exp/compat/auth/index.ts:3]: 2020 -> 2021 and elsewhere

File: packages-exp/firebase-exp/compat/empty-import.d.ts [hunk 3]
```diff
@@ -0,0 +1,20 @@
+/**
+ * @license
+ * Copyright 2018 Google LLC
```
Comment (hsubox76) [inline packages-exp/firebase-exp/compat/empty-import.d.ts:3]: 2018 -> 2021

File: packages-exp/firebase-exp/compat/index.ts [hunk 4]
```diff
@@ -39,15 +39,15 @@ import 'firebase/<PACKAGE>';
 import firebase from './app';
 import { name, version } from '../package.json';
 
-// import './auth';
+import './auth';
 // import './database';
 // import './firestore';
```
Comment (hsubox76) [inline packages-exp/firebase-exp/compat/index.ts:44]: firestore/storage being added in other PRs?

File: packages-exp/auth-compat-exp/index.ts (hunk 1 above)
Comment (Feiyang1) [reply packages-exp/auth-compat-exp/index.ts:143]: Auth-exp will register itself, so it's not necessary to do it in auth-compat

File: packages-exp/firebase-exp/compat/index.ts (hunk 4 above)
Comment (Feiyang1) [reply packages-exp/firebase-exp/compat/index.ts:44]: Good catch. added them here.

File: packages-exp/firebase-exp/compat/auth/index.ts (hunk 2 above)
Comment (Feiyang1) [reply packages-exp/firebase-exp/compat/auth/index.ts:3]: Updated and updated the license script.

**Review by hsubox76 — APPROVED** [review]

**Pull request merged**